service/worker/batcher/workflow.go

```diff
@@ -183,7 +183,7 @@ func setDefaultParams(params BatchParams) BatchParams {
 
 // BatchActivity is activity for processing batch operation
 func BatchActivity(ctx context.Context, batchParams BatchParams) (HeartBeatDetails, error) {
```

Comment:
Nonblocking decision, I think you have two methods of getting stuff into activites, there's the context.Value() method here (and used in other parts of cadence's workflows)

```
/ DisableArchivalActivity disables archival for the domain
func (w *domainDeprecator) DisableArchivalActivity(ctx context.Context, params DomainActivityParams) error {
	client := w.clientBean.GetFrontendClient() // you can pull stuff off the `w` and use it here, no need to use ctx.Value()
```

Generally the method approach like you're using in the disableArchivalActivity gives you easier development, because you can see what functions and data are available, they're not buried mysteriously somewhere on the context object. But this is just a suggestion.